src/olympia/devhub/tasks.py

```diff
@@ -388,24 +401,47 @@ def run_validator(path, for_appversions=None, test_all_tiers=False,
         return json_result
 
 
-def track_validation_stats(json_result):
+def run_addons_linter(path, listed=True):
+    from .utils import fix_addons_linter_output
+    process = subprocess.Popen([
+        settings.ADDONS_LINTER_BIN,
+        path,
+        '--boring'
+    ], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    with statsd.timer('devhub.linter'):
+        stdout, stderr = process.communicate()
+
+    if stderr:
+        raise ValueError(stderr)
+
+    parsed_data = json.loads(stdout)
+
+    result = json.dumps(fix_addons_linter_output(parsed_data, listed))
+    track_validation_stats(result, addons_linter=True)
+
+    return result
+
+
+def track_validation_stats(json_result, addons_linter=False):
     """
     Given a raw JSON string of validator results, log some stats.
     """
     result = json.loads(json_result)
```

Comment:
Unfortunately, yes. we're loading the json from stdout, fixing it up and dumping it again to make it work with all existing APIs.

Given that the validation results are usually very small it shouldn't be a problem.